catalog/rocks/catalog.go

```diff
@@ -104,17 +113,50 @@ type EntryRecord struct {
 
 // Commit represents commit metadata (author, time, tree ID)
 type Commit struct {
-	Committer    string
-	Message      string
-	TreeID       TreeID
-	CreationDate time.Time
-	Parents      []CommitID
-	Metadata     map[string]string
+	Committer    string           `db:"committer"`
+	Message      string           `db:"message"`
+	TreeID       TreeID           `db:"tree_id"`
+	CreationDate time.Time        `db:"creation_date"`
+	Parents      CommitParents    `db:"parents"`
+	Metadata     catalog.Metadata `db:"metadata"`
+}
+
+func (c Commit) ID() CommitID {
+	// TODO(ozkatz): move the hashing logic and encoding to a module others can use
+	h := sha256.New()
+	h.Write([]byte(c.Committer))
+	h.Write([]byte{0})
+	h.Write([]byte(c.Message))
+	h.Write([]byte{0})
+	h.Write([]byte(c.TreeID))
+	h.Write([]byte{0})
+	binary.Write(h, binary.BigEndian, c.CreationDate.Unix())
```

Comment:
Consider just writing ISO8601 format string: easier to track.

Reply:
started with that, got bitten by Postgres that converted the UTC date I inserted into my local timezone when I selected the record, resulting in a different hash than the one I used when storing the object. epoch is always UTC and it seems like it's what Git is using (according to `git cat-file -p ...` anyway)